### PR TITLE
Begin refactoring repeated vulkan low level code

### PR DESCRIPTION
**Pull request by acmcarther**
@seanstrom

### CODE REVIEW

File: client/presentation/BUILD [hunk 1]
```diff
@@ -21,24 +21,23 @@ rust_library(
     srcs = ["sdl2_vulkan_interop.rs"],
     deps = [
         ":vulkan",
-        "//third_party/SDL2:sdl2_sys",
         "//cargo:sdl2",
         "//cargo:vk_sys",
+        "//cargo/overrides/sdl2-sys:sdl2_sys",
```
Comment (wmedrano) [inline client/presentation/BUILD]: what happened to third party?

File: client/presentation/vulkan.rs [hunk 2]
```diff
@@ -235,29 +246,191 @@ impl VkCtx {
...
+    let mut num_physical_devices = 0u32;
+    unsafe {
+      let result = self.instance_ptrs(instance).EnumeratePhysicalDevices(
+        instance, &mut num_physical_devices, ptr::null_mut());
+
+      if result != vk::SUCCESS {
+        return Err(VkRawReturnCode(result as i32, "while enumerating physical devices".to_owned()));
+      }
+
+      println!("found {} physical devices", num_physical_devices);
```
Comment (wmedrano) [inline client/presentation/vulkan.rs]: wtf

File: client/presentation/vulkan.rs [hunk 3]
```diff
@@ -49,6 +53,15 @@ struct SwapchainParams {
   extent: vk::Extent2D,
 }
 
+macro_rules! do_or_die {
+  ($res:expr) => {
+    match $res {
+      Err(VkRawReturnCode(code, ctx_string)) => panic!("Low level error {} with context: {}", vk_result_to_human(code), ctx_string),
```
Comment (wmedrano) [inline client/presentation/vulkan.rs]: Just use .unwrap() and implement debug if you want better debug string

File: WORKSPACE [hunk 4]
```diff
@@ -56,9 +56,17 @@ raze_fetch_remote_crates()
 new_local_repository(
     name = "llvm",
     build_file = "llvm.BUILD",
-    path = "/usr/lib/llvm-3.9",
+    path = "/usr/lib",
```
Comment (wmedrano) [inline WORKSPACE]: wtf

File: client/presentation/BUILD (hunk 1 above)
Comment (acmcarther) [reply client/presentation/BUILD]: I moved the bindgen invocation and rust_library out of third_party (where it kind-of didn't belong), and put it in cargo/overrides where people would expect it to be.

As a recap, this is a me-generated override for sdl2-sys.

**Comment by acmcarther** [comment]
This PR has become overgrown, merging.

**Pull request merged**